### PR TITLE
Integrated Working Directory to application

### PR DESCRIPTION
**Pull request by AlmightyLks**
As suggested in #191 , here goes the integration of the Working Directory option.  

Automatically set by choosing the executable, fallback if null (e.g. when not specified within the CLI), also getting the working directory from the executable path.

Only "problem": 
Localization. I don't know any of the languages, especially uwu 😜, how do you usually approach this?

### CODE REVIEW

**Comment by AlmightyLks** [comment]
Preview:
image

image

File: source/Reloaded.Mod.Launcher.Lib/Misc/Constants.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ public static class Constants
     public const string ParameterKill = "--kill";
     public const string ParameterLaunch = "--launch";
     public const string ParameterArguments = "--arguments";
+    public const string WorkingDirectory = "--working-directory";
```
Comment (Sewer56) [inline source/Reloaded.Mod.Launcher.Lib/Misc/Constants.cs:12]: This should probably be renamed to match the rest of the CLI parameters.

File: source/Reloaded.Mod.Launcher/Assets/Languages/en-GB.xaml [hunk 2]
```diff
@@ -36,6 +36,7 @@
     <sys:String x:Key="AddAppName">Application Name</sys:String>
     <sys:String x:Key="AddAppLocation">Executable Location</sys:String>
     <sys:String x:Key="AddAppArguments">Commandline Arguments</sys:String>
+    <sys:String x:Key="AddAppWorkingDirectory">Working Directory</sys:String>
```
Comment (Sewer56) [inline source/Reloaded.Mod.Launcher/Assets/Languages/en-GB.xaml:39]: Please move this one to the bottom of the file, and add next version number to it (1.23.8);
at one point I moved to using chronological order to make localisation easier.

File: source/Reloaded.Mod.Launcher.Lib/Misc/Constants.cs (hunk 1 above)
Comment (AlmightyLks) [reply source/Reloaded.Mod.Launcher.Lib/Misc/Constants.cs:12]: whoops

**Comment by Sewer56** [comment]
Looks good to me 👍

File: source/Reloaded.Mod.Launcher.Lib/Startup.cs [hunk 3]
```diff
@@ -84,16 +84,18 @@ private static void LaunchApplicationAndExit(string applicationToLaunch)
         if (application != null)
             arguments = $"{arguments} {application.Config.AppArguments}";
 
+        _commandLineArguments.TryGetValue(Constants.ParameterArguments, out var workingDirectory);
```
Comment (Sewer56) [inline source/Reloaded.Mod.Launcher.Lib/Startup.cs:87]: Typo here, wrong parameter referenced

Comment (AlmightyLks) [reply source/Reloaded.Mod.Launcher.Lib/Startup.cs:87]: !

**Comment by Sewer56** [comment]
I think after this fixup, should be good to go; just run from CLI with custom working dir and see if you're getting the results you're expecting.

**Comment by AlmightyLks** [comment]
Quick example on using the CLI?

**Comment by Sewer56** [comment]
`Reloaded-II.exe --launch "PATH_TO_GAME_EXE"`

If in doubt, hit `Create Shortcut` in launcher and modify whatever that spits out.

**Comment by AlmightyLks** [comment]
Found another bug, gonna push a fix

But besides that, funny:  
a cli argument of `--launch "C:\Program Files (x86)\AssaultCube 1.3.0.2\bin_win32\ac_client.exe" --working-directory "C:\Program Files (x86)\AssaultCube 1.3.0.2\bin_win32\"` will result in the very last argument's backslash for the directory indication interferes with the closing parenthesis, and it is seen as an escaped parenthesis
image

**Comment by Sewer56** [comment]
Creating processes can be real tricky ahahaha, yeah :p

**Comment by AlmightyLks** [comment]
And MS states, that this is not a bug
https://developercommunity.visualstudio.com/t/environmentgetcommandlineargs-and-myapplicationcom/473717#T-N503552

**Comment by Sewer56** [comment]
Usually you'd just want to trim the last `/` or `\` character if present instead; some APIs don't allow directories with ending backslash anyway.

**Comment by AlmightyLks** [comment]
Oh really?
I would've assume that it is the go-to standard to differentiate between files and folders

**Comment by Sewer56** [comment]
Yeah; that's been my experience with some things.
Personally I do prefer leaving a backslash though to indicate intent.

**Comment by AlmightyLks** [comment]
Both CLI and GUI are working in this state as of now
Ready for merge 😄

**Comment by Sewer56** [comment]
Should be live in a few minutes.

**Comment by AlmightyLks** [comment]
No hurry
It's not something I personally need in order to do something special

You can gladly keep it until it's worth another release

**Comment by Sewer56** [comment]
I probably wouldn't be doing maintenance on Reloaded itself for a while (except for libraries), so it's better for it not to sit unreleased :p

**Pull request merged**